src/types/core/key.rs

```diff
@@ -90,14 +95,45 @@ impl Display for Version {
     }
 }
 
+impl SerializeValue for Version {
+    fn serialize<'b>(
+        &self,
+        typ: &ColumnType<'_>,
+        writer: scylla::serialize::writers::CellWriter<'b>,
+    ) -> Result<
+        scylla::serialize::writers::WrittenCellProof<'b>,
+        scylla::serialize::SerializationError,
+    > {
+        if typ != &scylla::frame::response::result::ColumnType::Int {
```

Comment:
how does the type get inferred for serialize, do we need this check

Reply:
Not needed this check would happen in the serde deserialization itself